app/models/grda_warehouse/tasks/service_history/enrollment.rb

```diff
@@ -606,27 +606,32 @@ def entry_exit_tracking?
     end
 
     def street_outreach_acts_as_bednight?
-      @street_outreach_acts_as_bednight ||= if project.so?
-          project.services.where(
-          Services: {RecordType: 12},
-        ).exists?
-        else
-          false
-        end
+      @street_outreach_acts_as_bednight ||= project.so? && project.
+        joins(enrollments: :current_living_situations).exists?
+    end
+
+    def project_extrapolates_contacts?
+      project.extraplolate_contacts
     end
 
     def build_for_dates
-      @build_for_dates ||= begin
-        if entry_exit_tracking?
-          (self.EntryDate..build_until).map do |date|
-            [date, service_type_from_project_type(project.computed_project_type)]
-          end.to_h
-        else
-          # Fetch all services provided between the start of the enrollment and the end of the build period
-          services.where(DateProvided: (self.EntryDate..build_until)).
-            order(DateProvided: :asc).
-            pluck(:DateProvided, :TypeProvided).to_h
+      @build_for_dates ||= if entry_exit_tracking?
+        (self.EntryDate..build_until).map do |date|
+          [date, service_type_from_project_type(project.computed_project_type)]
+        end.to_h
+      else
+        # Fetch all services provided between the start of the enrollment and the end of the build period
+        service_records = services.where(DateProvided: (self.EntryDate..build_until)).
+          order(DateProvided: :asc).
+          pluck(:DateProvided, :TypeProvided).to_h
+        if project.so?
+          # Find all contacts for SO.  Pretend like they are bed-nights
+          living_situations = current_living_situations.where(InformationDate: (self.EntryDate..build_until)).
+            order(InformationDate: :asc).
+            pluck(:InformationDate).map{ |d| [d, 200] }.to_h
```

Comment:
Why 200?

Reply:
`200 => 'Bed night',`  Might not be 100% correct, but I'm not sure what else to put in there.